src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -104,31 +117,24 @@ protected Optional<TokenPermissions> getTokenPermissions(HttpServletRequest requ
         return AuthorisationUtil.getTokenPermissions(request);
     }
 
-    private boolean doRoeUpdateJourneyChecks(HttpServletRequest request, HttpServletResponse response, String reqId, String companyNumberInScope,
-                                             String transactionId, Map<String, Object> authInfoMap) {
-        boolean isValidOeNumber = companyNumberInScope.startsWith("OE");
-        if (isValidOeNumber) {
-            boolean doCompanyNumbersMatch = doCompanyNumbersMatch(request, response, companyNumberInScope, transactionId);
-            authInfoMap.put("roe_update_journey_company_numbers_match", doCompanyNumbersMatch);
-            if (doCompanyNumbersMatch) {
-                ApiLogger.debugContext(reqId, "UserAuthenticationInterceptor authorised with company_oe_annual_update=create permission",
-                        authInfoMap);
-                return true;
-            } else {
-                ApiLogger.errorContext(reqId, "UserAuthenticationInterceptor unauthorised for ROE Update Journey as Company Numbers inScope and Transaction do not match", null, authInfoMap);
-                return false;
-            }
-        } else {
-            ApiLogger.debugContext(reqId, "UserAuthenticationInterceptor unauthorised for ROE Update Journey as Company Number is not a valid OE",
-                    authInfoMap);
-            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-            return false;
+    private boolean isRequestRoeRegistration(Optional<String> companyNumberInTransaction) {
+        return companyNumberInTransaction.isEmpty();
+    }
+
+    private boolean isRequestRoeUpdate(String companyNumberInScope, Optional<String> companyNumberInTransaction) {
+        boolean isValidOeNumberInScope = isValidOeNumber(companyNumberInScope);
+        var isValidOeNumberInTransaction = false;
+        if (companyNumberInTransaction.isPresent()) {
+            isValidOeNumberInTransaction = isValidOeNumber(companyNumberInTransaction.get());
+        }
+        if (isValidOeNumberInScope && isValidOeNumberInTransaction) {
+            return companyNumberInScope.equalsIgnoreCase(companyNumberInTransaction.get());
         }
+        return false;
     }
 
-    private boolean isRequestRoeUpdate(String companyNumberInScope) {
-        // Further check for OE number is done in doRoeUpdateJourneyChecks method
-        return StringUtils.isNotEmpty(companyNumberInScope);
+    private boolean isValidOeNumber(String oeNumber) {
+        return StringUtils.isNotEmpty(oeNumber) && oeNumber.startsWith("OE");
```

Comment:
I'd be tempted to define a class-level constant named something like `OVERSEAS_ENTITY_COMPANY_NUMBER_PREFIX` for the "OE" string and use that here.